### PR TITLE
upgrade to Lumen 7 and neomerx/json-api 4.*

### PR DESCRIPTION
**Pull request by avanca-ro**

### CODE REVIEW

File: src/EncoderService.php [hunk 1]
```diff
@@ -74,6 +78,11 @@ protected function getEncoderOptions(array $config)
         $depth = isset($config['depth']) && is_int($config['depth']) ?
             $config['depth'] :
             512;
-        return new EncoderOptions($options, $urlPrefix, $depth);
+
```
Comment (pghiuzan-ro) [inline src/EncoderService.php:81]: Is it necessary to avoid using the `EncoderOptions` class?

File: tests/EncoderServiceTest.php [hunk 2]
```diff
@@ -6,9 +6,11 @@
 
 class EncoderServiceTest extends \PHPUnit\Framework\TestCase
 {
-    public function setUp()
+    private $encoder_service;
```
Comment (pghiuzan-ro) [inline tests/EncoderServiceTest.php:9]: Use camelCase names, please.

File: tests/EncoderServiceTest.php [hunk 3]
```diff
@@ -67,9 +69,9 @@ public function testGetEncoderOptionsDefaults()
 
         $encoder_options = $method->invokeArgs($service, [[]]);
```
Comment (pghiuzan-ro) [inline tests/EncoderServiceTest.php:70]: Camel case

File: src/EncoderService.php (hunk 1 above)
Comment (avanca-ro) [reply src/EncoderService.php:81]: It's been removed from the package.

File: tests/Middleware/EnforceMediaTypeTest.php [hunk 4]
```diff
@@ -50,7 +50,7 @@ public function testHandlesIncorrectHeadersForData()
 
     public function testHandlesIncorrectlySetAcceptHeader()
     {
-        $guard   = $this->getMockBuilder(MediaTypeGuard::class)->disableOriginalConstructor()->setMethods([
+        $guard   = $this->getMockBuilder(MediaTypeGuard::class)->disableOriginalConstructor()->onlyMethods([
```
Comment (pghiuzan-ro) [inline tests/Middleware/EnforceMediaTypeTest.php:53]: What's the reason behind changing `setMethods` with `onlyMethods` everywhere?

Comment (avanca-ro) [reply tests/Middleware/EnforceMediaTypeTest.php:53]: setMethods() is deprecated now

File: src/EncoderService.php (hunk 1 above)
Comment (pghiuzan-ro) [reply src/EncoderService.php:81]: Got it. Thanks.

File: src/Requests/MalformedRequest.php [hunk 5]
```diff
@@ -20,16 +20,27 @@ public function getId()
...
+        return null;
     }
 
     /**
      * Get the HTTP status code applicable to this problem, expressed as a string value.
      *
      * @return string|null
      */
-    public function getStatus()
+    public function getStatus(): string
```
Comment (spavel-ro) [inline src/Requests/MalformedRequest.php:43]: This method has to return `string` but the actual return value is int

File: tests/Middleware/EnforceMediaTypeTest.php (hunk 4 above)
Comment (pghiuzan-ro) [reply tests/Middleware/EnforceMediaTypeTest.php:53]: Got it. Thanks.

**Pull request merged**